### DISCUSSION TITLE
How to rotate both appearance and layout

### DISCUSSION THREAD

**Discussion by uben0** [typst/typst#2274] [Q&A] [answered]
```typst
#table(rotate(90deg)[hello])
```
The `rotate` function only rotates the appearance but not the layout. Is there a way to write vertical text and have it occupying the same space as its appearance?

**Comment by 8LWXpg** [comment, marked as answer]
make a box with reversed height and width:

```typst
#set box(stroke: 0.5pt+luma(200))
#let rotate-layout(rot, body) = style(styles => {
  let size = measure(body, styles)
  box(width: size.height, height: size.width, rotate(rot)[#body])
})

aaa#box(rotate(90deg)[hello])aaa

aaa#box(rotate-layout(90deg)[hello])aaa
```

<img width="131" alt="image" src="https://github.com/typst/typst/assets/105704427/d42f9dfd-cf09-4cc9-aa9a-c4e316e01db1">

**Reply by uben0** [reply]
I can even see a version accepting any rotation and using sinus and cosines to determine width and height
